### ISSUE TITLE
DWF IDs for potential kernel issues

### ISSUE THREAD

**Issue by joshbressers** [distributedweaknessfiling/dwf-workflow#31]
DWF and the Linux Kernel have had a discussion regarding assigning IDs for potential issues in the Kernel. The kernel has had issues with the legacy CVE system and gave up on acquiring IDs. Given the flexibility of DWF they are interested in requesting IDs for problems.

This issue is to capture some of the discussion and purpose behind what's happening in this space.

The conversation started with this note from greg k-h

> But really, a way for me to assign a CVE to pretty much every stable
> kernel commit that is made, that we know fixes a real issue, would be
> nice to do. 
> 
> ...
> 
> So, any way I can automate this through the DWF project?

Sasha Levin then suggested this format

> Just to get an idea of what we're looking for, how about this as a base:
> 
> 1aa561b1a4c0ae2a9a9b9c21a84b5ca66b4775d8,v5.9,04c4f2ee3f68c9a4bf1653d15f1a9a435ae33f7a,v5.11.16
> 
> Where the format is:
> 
> <commit introduced bug>,<version introduced>,<commit that fixes the issue>,<fixed version>

The decision was made that one ID per tree would be best for the purposes of tracking these commits.

**Comment by kurtseifried** [comment]
Yeah it's pretty simple, the Linux Kernel knows best what is/isn't a security issue, they get IDs, and if they turn out to not be a security vulnerability we can mark them as not a vulnerability. If they are a vulnerability they can be updated/expanded with vendor information and so on as usual. I'd much rather have eyeballs on potential security vulnerabilities than ignore stuff that later turns out to be a problem.

**Comment by joshbressers** [comment]
The script used for this task can be found here
https://github.com/distributedweaknessfiling/dwf-request/blob/main/dwf-bot/helpers/linux-kernel-generator.py
